### PR TITLE
ENH: Add DOC commit comment prefix to commit message check hook

### PR DESCRIPTION
**Pull request by lassoan**

### CODE REVIEW

File: commit-msg [hunk 1]
```diff
@@ -91,10 +92,14 @@ Valid commit types are:
...
   COMP:  - a fix for a compilation issue, error or warning,
+  DOC:   - a change made to documentation files,
   ENH:   - new functionality added to the project,
   PERF:  - a performance improvement,
-  STYLE: - a change that does not impact the logic or execution of the code.
-  (improve coding style, comments, documentation).
+  STYLE: - a code change that does not impact the logic or execution of the code.
+  (improve coding style, comments, documentation in source code).
+
+Also consider adding reference to related issue by adding "re #123" (without quotes,
```
Comment (jcfr) [inline commit-msg:101]: Or `See #123`

Comment (jcfr) [inline commit-msg:101]:
```suggestion
Also consider adding reference to related issue by adding "re #123" (without quotes,
Also consider adding reference to related issue by adding "See #123" (without quotes,
```

**Comment by jamesobutler** [comment]
Does there still need to be git hook check for commit message prefix when there is already the usage of the CommitCheck GitHub app on PRs?

**Comment by cpinter** [comment]
I think it is useful to be able to catch this as early as possible. Then the contributor can fix it locally before pushing

**Comment by lassoan** [comment]
Thank you, committed with the suggested changes.

**Pull request closed without merging**